inc/namespace.php

```diff
@@ -43,6 +45,37 @@ function admin_bar_render() {
 	die();
 }
 
+/**
+ * Set a cookie to identify that the current user is logged in.
+ *
+ * We can't rely on WordPress's LOGGED_IN_COOKIE, which is HTTP-only.
+ */
+function set_js_cookie() {
+	setcookie(
+		'wp_remote_admin_bar',
+		1,
+		[
+			'domain'   => COOKIEDOMAIN,
+			'path'     => SITE_COOKIE_PATH,
+			'httponly' => false
```

Comment:
Without an `expiry` here this will be a session cookie, which means a user can log in, close their browser, open it up again, and still be logged in but not see the admin bar because this cookie will be deleted.

This should probably use the same expiry time as the auth cookie, which can be gotten via the `auth_cookie_expiration` filter:

```php
$expiry = apply_filters( 'auth_cookie_expiration', 172800, get_current_user_id(), false );
```

Reply:
It's surprisingly hard to find out how long a admin cookie is valid for. I ended up using the higher value of 14 days, which is the default if the user selects "Remember me". If this cookie is valid but the user isn't still logged in, then there's no real damage done other than making unnecessary ajax requests. But if the remote admin bar cookie expires before the auth cookie, then we wouldn't even be checking for an admin bar, which seems worse.